### ISSUE TITLE
Do you have plans to respond to the latest changes at Travis?

### ISSUE THREAD

**Issue by tomjaguarpaw** [ndmitchell/neil#50]
travis-ci.org is shutting down. Open source projects can still get free CI minutes on travis-ci.com but I get the impression that the quality of the free service is seriously degrading. See https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works for some details.

It's possible that I have few enough repos that the free travis-ci.com repos will still cover me. If not I may just subscribe to the paid service rather than spend time migrating away.

@ndmitchell I was wondering if you had concrete plans about this.

**Comment by tomjaguarpaw** [comment]
> I may just subscribe to the paid service

I discovered that [the cheapest plan is $69 a month](https://travis-ci.com/plans) so, no, I won't be subscribing :)

**Comment by ndmitchell** [comment]
I wasn't aware - thanks for the heads up. 1000 minutes a month is nowhere near enough for me. Looks like github actions looms in my future...

**Comment by ndmitchell** [comment]
There is now a github actions file, which I expect to be refining over the next few days and rolling out to my projects.

**Comment by ndmitchell** [comment]
I've now rolled it out to all my projects. There is a lot that could be improved, e.g.:

* I'd like to wait for movement on https://github.com/actions/runner/issues/646 so I can collapse a lot of the actions (e.g. setting up the cache) into the action.
* The `neil` binary is a bit out of control. Perhaps it should just be a Haskell script which does a lot less.
* Using `cabal new-foo` style commands means that copying to a separate temp directory isn't required. That's a lot of the complexity.
* Maybe the checks should be disjoint from the execution, so I build using normal cabal new-install etc, but then run `check hoogle`, `check repo` etc separately.

However, the speed is quite good, so I'm very happy for now. Closing this issue.

In the transition its entirely possible I broke a few of your CI things using my script. If so, happy to get a patch to put them back.
